src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,243 @@
+  groupId?: string;
+  groupDisplayName?: string;
+  role?: string;
+  properties?: string;
+  filter?: string;
+}
+
+interface ExtendedUser extends User {
+  roles: string[];
+}
+
+class AadGroupUserListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.GROUP_USER_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists users of a specific Azure AD group';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'userPrincipalName', 'roles'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        role: typeof args.options.role !== 'undefined',
+        properties: typeof args.options.properties !== 'undefined',
+        filter: typeof args.options.filter !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member"]
+      },
+      {
+        option: "-p, --properties [properties]"
+      },
+      {
+        option: "-f, --filter [filter]"
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+
+      let users: User[] = [];
+
+      switch (args.options.role) {
+        case 'Owner':
+          users = await this.getOwners(args.options, groupId, logger);
+          break;
+        case 'Member':
+          users = await this.getMembers(args.options, groupId, logger);
+          break;
+        default:
+          const owners = await this.getOwners(args.options, groupId, logger);
+          const members = await this.getMembers(args.options, groupId, logger);
+
+          if (!args.options.properties) {
```

Comment:
Why would you only do this if properties is not used? Probably because the userPrincipalName might not be available. 

For that reason, I think we'll need to compare on `id`, and make sure `id` is always forced when `$select` is used. I think that is understandable behavior.